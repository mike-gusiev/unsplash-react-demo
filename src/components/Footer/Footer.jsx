import React from 'react';
import { Row } from 'react-bootstrap'
import { SocialIcon } from 'react-social-icons';

import './Footer.scss';

export const Footer = () => {
  return (
    <>
      <Row className="social-icon-list">
        <SocialIcon url="http://twitter.com" />
        <SocialIcon url="http://facebook.com" />
        <SocialIcon url="http://youtube.com" />
      </Row>
      <Row className="unsplash-copyright">
        Â© 2019 Unsplash, Inc.
      </Row>
    </>
  )
};
