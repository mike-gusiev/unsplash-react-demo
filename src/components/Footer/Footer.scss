.social-icon-list {
  display: flex;
  justify-content: space-around;
  padding: 12px 0;
}

.unsplash-copyright {
  text-align: center;
  padding: 12px 0;
}
