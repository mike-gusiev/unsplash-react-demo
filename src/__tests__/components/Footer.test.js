import React from 'react';
import rendered from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Footer } from '../../components/Footer/Footer';

configure({adapter: new Adapter()});


describe('Footer component', () => {

    test('matches the snapshot', () => {
        const tree = rendered.create(<Router><Footer/></Router>).toJSON();
        expect(tree).toMatchSnapshot();
    });

});
