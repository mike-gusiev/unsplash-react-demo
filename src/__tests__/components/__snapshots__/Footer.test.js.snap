// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component matches the snapshot 1`] = `
Array [
  <div
    className="social-icon-list row"
  >
    <a
      aria-label="twitter"
      className="social-icon"
      href="http://twitter.com"
      style={
        Object {
          "display": "inline-block",
          "height": "50px",
          "overflow": "hidden",
          "position": "relative",
          "verticalAlign": "middle",
          "width": "50px",
        }
      }
    >
      <div
        className="social-container"
        style={
          Object {
            "height": "100%",
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
      >
        <svg
          className="social-svg"
          style={
            Object {
              "borderRadius": "50%",
              "fillRule": "evenodd",
              "height": "100%",
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": "100%",
            }
          }
          viewBox="0 0 64 64"
        >
          <g
            className="social-svg-background"
            style={
              Object {
                "MozTransition": "fill 170ms ease-in-out",
                "OTransition": "fill 170ms ease-in-out",
                "WebkitTransition": "fill 170ms ease-in-out",
                "fill": "transparent",
                "msTransition": "fill 170ms ease-in-out",
                "transition": "fill 170ms ease-in-out",
              }
            }
          >
            <circle
              cx="32"
              cy="32"
              r="31"
            />
          </g>
          <g
            className="social-svg-icon"
            style={
              Object {
                "MozTransition": "fill 170ms ease-in-out",
                "OTransition": "fill 170ms ease-in-out",
                "WebkitTransition": "fill 170ms ease-in-out",
                "fill": "transparent",
                "msTransition": "fill 170ms ease-in-out",
                "transition": "fill 170ms ease-in-out",
              }
            }
          >
            <path
              d="M48,22.1c-1.2,0.5-2.4,0.9-3.8,1c1.4-0.8,2.4-2.1,2.9-3.6c-1.3,0.8-2.7,1.3-4.2,1.6 C41.7,19.8,40,19,38.2,19c-3.6,0-6.6,2.9-6.6,6.6c0,0.5,0.1,1,0.2,1.5c-5.5-0.3-10.3-2.9-13.5-6.9c-0.6,1-0.9,2.1-0.9,3.3 c0,2.3,1.2,4.3,2.9,5.5c-1.1,0-2.1-0.3-3-0.8c0,0,0,0.1,0,0.1c0,3.2,2.3,5.8,5.3,6.4c-0.6,0.1-1.1,0.2-1.7,0.2c-0.4,0-0.8,0-1.2-0.1 c0.8,2.6,3.3,4.5,6.1,4.6c-2.2,1.8-5.1,2.8-8.2,2.8c-0.5,0-1.1,0-1.6-0.1c2.9,1.9,6.4,2.9,10.1,2.9c12.1,0,18.7-10,18.7-18.7 c0-0.3,0-0.6,0-0.8C46,24.5,47.1,23.4,48,22.1z"
            />
          </g>
          <g
            className="social-svg-mask"
            style={
              Object {
                "MozTransition": "fill 170ms ease-in-out",
                "OTransition": "fill 170ms ease-in-out",
                "WebkitTransition": "fill 170ms ease-in-out",
                "fill": "#00aced",
                "msTransition": "fill 170ms ease-in-out",
                "transition": "fill 170ms ease-in-out",
              }
            }
          >
            <path
              d="M0,0v64h64V0H0z M44.7,25.5c0,0.3,0,0.6,0,0.8C44.7,35,38.1,45,26.1,45c-3.7,0-7.2-1.1-10.1-2.9 c0.5,0.1,1,0.1,1.6,0.1c3.1,0,5.9-1,8.2-2.8c-2.9-0.1-5.3-2-6.1-4.6c0.4,0.1,0.8,0.1,1.2,0.1c0.6,0,1.2-0.1,1.7-0.2 c-3-0.6-5.3-3.3-5.3-6.4c0,0,0-0.1,0-0.1c0.9,0.5,1.9,0.8,3,0.8c-1.8-1.2-2.9-3.2-2.9-5.5c0-1.2,0.3-2.3,0.9-3.3 c3.2,4,8.1,6.6,13.5,6.9c-0.1-0.5-0.2-1-0.2-1.5c0-3.6,2.9-6.6,6.6-6.6c1.9,0,3.6,0.8,4.8,2.1c1.5-0.3,2.9-0.8,4.2-1.6 c-0.5,1.5-1.5,2.8-2.9,3.6c1.3-0.2,2.6-0.5,3.8-1C47.1,23.4,46,24.5,44.7,25.5z"
            />
          </g>
        </svg>
      </div>
    </a>
    <a
      aria-label="facebook"
      className="social-icon"
      href="http://facebook.com"
      style={
        Object {
          "display": "inline-block",
          "height": "50px",
          "overflow": "hidden",
          "position": "relative",
          "verticalAlign": "middle",
          "width": "50px",
        }
      }
    >
      <div
        className="social-container"
        style={
          Object {
            "height": "100%",
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
      >
        <svg
          className="social-svg"
          style={
            Object {
              "borderRadius": "50%",
              "fillRule": "evenodd",
              "height": "100%",
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": "100%",
            }
          }
          viewBox="0 0 64 64"
        >
          <g
            className="social-svg-background"
            style={
              Object {
                "MozTransition": "fill 170ms ease-in-out",
                "OTransition": "fill 170ms ease-in-out",
                "WebkitTransition": "fill 170ms ease-in-out",
                "fill": "transparent",
                "msTransition": "fill 170ms ease-in-out",
                "transition": "fill 170ms ease-in-out",
              }
            }
          >
            <circle
              cx="32"
              cy="32"
              r="31"
            />
          </g>
          <g
            className="social-svg-icon"
            style={
              Object {
                "MozTransition": "fill 170ms ease-in-out",
                "OTransition": "fill 170ms ease-in-out",
                "WebkitTransition": "fill 170ms ease-in-out",
                "fill": "transparent",
                "msTransition": "fill 170ms ease-in-out",
                "transition": "fill 170ms ease-in-out",
              }
            }
          >
            <path
              d="M34.1,47V33.3h4.6l0.7-5.3h-5.3v-3.4c0-1.5,0.4-2.6,2.6-2.6l2.8,0v-4.8c-0.5-0.1-2.2-0.2-4.1-0.2 c-4.1,0-6.9,2.5-6.9,7V28H24v5.3h4.6V47H34.1z"
            />
          </g>
          <g
            className="social-svg-mask"
            style={
              Object {
                "MozTransition": "fill 170ms ease-in-out",
                "OTransition": "fill 170ms ease-in-out",
                "WebkitTransition": "fill 170ms ease-in-out",
                "fill": "#3b5998",
                "msTransition": "fill 170ms ease-in-out",
                "transition": "fill 170ms ease-in-out",
              }
            }
          >
            <path
              d="M0,0v64h64V0H0z M39.6,22l-2.8,0c-2.2,0-2.6,1.1-2.6,2.6V28h5.3l-0.7,5.3h-4.6V47h-5.5V33.3H24V28h4.6V24 c0-4.6,2.8-7,6.9-7c2,0,3.6,0.1,4.1,0.2V22z"
            />
          </g>
        </svg>
      </div>
    </a>
    <a
      aria-label="youtube"
      className="social-icon"
      href="http://youtube.com"
      style={
        Object {
          "display": "inline-block",
          "height": "50px",
          "overflow": "hidden",
          "position": "relative",
          "verticalAlign": "middle",
          "width": "50px",
        }
      }
    >
      <div
        className="social-container"
        style={
          Object {
            "height": "100%",
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
      >
        <svg
          className="social-svg"
          style={
            Object {
              "borderRadius": "50%",
              "fillRule": "evenodd",
              "height": "100%",
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": "100%",
            }
          }
          viewBox="0 0 64 64"
        >
          <g
            className="social-svg-background"
            style={
              Object {
                "MozTransition": "fill 170ms ease-in-out",
                "OTransition": "fill 170ms ease-in-out",
                "WebkitTransition": "fill 170ms ease-in-out",
                "fill": "transparent",
                "msTransition": "fill 170ms ease-in-out",
                "transition": "fill 170ms ease-in-out",
              }
            }
          >
            <circle
              cx="32"
              cy="32"
              r="31"
            />
          </g>
          <g
            className="social-svg-icon"
            style={
              Object {
                "MozTransition": "fill 170ms ease-in-out",
                "OTransition": "fill 170ms ease-in-out",
                "WebkitTransition": "fill 170ms ease-in-out",
                "fill": "transparent",
                "msTransition": "fill 170ms ease-in-out",
                "transition": "fill 170ms ease-in-out",
              }
            }
          >
            <path
              d="M46.7,26c0,0-0.3-2.1-1.2-3c-1.1-1.2-2.4-1.2-3-1.3C38.3,21.4,32,21.4,32,21.4h0 c0,0-6.3,0-10.5,0.3c-0.6,0.1-1.9,0.1-3,1.3c-0.9,0.9-1.2,3-1.2,3S17,28.4,17,30.9v2.3c0,2.4,0.3,4.9,0.3,4.9s0.3,2.1,1.2,3 c1.1,1.2,2.6,1.2,3.3,1.3c2.4,0.2,10.2,0.3,10.2,0.3s6.3,0,10.5-0.3c0.6-0.1,1.9-0.1,3-1.3c0.9-0.9,1.2-3,1.2-3s0.3-2.4,0.3-4.9 v-2.3C47,28.4,46.7,26,46.7,26z M28.9,35.9l0-8.4l8.1,4.2L28.9,35.9z"
            />
          </g>
          <g
            className="social-svg-mask"
            style={
              Object {
                "MozTransition": "fill 170ms ease-in-out",
                "OTransition": "fill 170ms ease-in-out",
                "WebkitTransition": "fill 170ms ease-in-out",
                "fill": "#ff3333",
                "msTransition": "fill 170ms ease-in-out",
                "transition": "fill 170ms ease-in-out",
              }
            }
          >
            <path
              d="M0,0v64h64V0H0z M47,33.1c0,2.4-0.3,4.9-0.3,4.9s-0.3,2.1-1.2,3c-1.1,1.2-2.4,1.2-3,1.3 C38.3,42.5,32,42.6,32,42.6s-7.8-0.1-10.2-0.3c-0.7-0.1-2.2-0.1-3.3-1.3c-0.9-0.9-1.2-3-1.2-3S17,35.6,17,33.1v-2.3 c0-2.4,0.3-4.9,0.3-4.9s0.3-2.1,1.2-3c1.1-1.2,2.4-1.2,3-1.3c4.2-0.3,10.5-0.3,10.5-0.3h0c0,0,6.3,0,10.5,0.3c0.6,0.1,1.9,0.1,3,1.3 c0.9,0.9,1.2,3,1.2,3s0.3,2.4,0.3,4.9V33.1z M28.9,35.9l8.1-4.2l-8.1-4.2L28.9,35.9z"
            />
          </g>
        </svg>
      </div>
    </a>
  </div>,
  <div
    className="unsplash-copyright row"
  >
    © 2019 Unsplash, Inc.
  </div>,
]
`;
